src/main/java/org/galaxyproject/dockstore_galaxy_interface/language/GalaxyWorkflowPlugin.java

```diff
@@ -61,11 +66,26 @@ public String launchInstructions(String trsID) {
     }
 
     @Override
-    public List<RowData> generateToolsTable(
-        String initialPath,
-        String contents,
-        Map<String, Pair<String, GenericFileType>> indexedFiles) {
-      throw new UnsupportedOperationException();
+    public List<RowData> generateToolsTable(String initialPath, String contents, Map<String, Pair<String, GenericFileType>> indexedFiles) {
+      final Map<String, Object> workflow = loadWorkflow(contents);
+      final Map<String, Object> elements = Cytoscape.getElements(workflow);
+      final List<Map> nodes = (List<Map>)elements.getOrDefault("nodes", Lists.newArrayList());
+      return nodes.stream().map(node -> {
+        final RowData rowData = new RowData();
+        final Map<String, Object> nodeData = (Map<String, Object>)node.get("data");
```

Comment:
Probably so, but just checking -- is `nodeData` guaranteed to not be `null`?

Reply:
Looks like yes
https://github.com/galaxyproject/dockstore-galaxy-interface/pull/10/files#diff-67da40517c8bf816074537fcf24ef99cR81